tasks/notify.py

```diff
@@ -443,6 +443,45 @@ def send_notification(ctx: Context, alert_jobs):
         send_slack_message("#agent-platform-ops", message)
 
 
+@task
+def send_failure_summary_notification(_, list_max_len=10):
+    jobs = os.environ["JOB_FAILURES"]
+    jobs = json.loads(jobs)
+
+    # List of (job_name, (failure_count, total_count)) ordered by failure_count
+    stats = sorted(
+        ((name, (fail, success)) for (name, (fail, success)) in jobs.items() if fail > 0),
+        key=lambda x: (x[1][0], x[1][1] if x[1][1] is not None else 0),
+        reverse=True,
+    )[:list_max_len]
+
+    # Don't send message if no failure
+    if len(stats) == 0:
+        return
+
+    # Create message
+    message = ['*Daily Job Failure Report*']
+    message.append('These jobs had the most failures in the last 24 hours:')
+    for name, (fail, total) in stats:
+        link = CI_VISIBILITY_JOB_URL.format(name.replace(' ', '%20'))
```

Comment:
instead of doing the replace yourself you could maybe store the url with "normal" text and use `urllib.parse.quote` to generate the correct value, wdyt?